library/lib/serve.js

```diff
@@ -135,9 +137,14 @@ function possibleDirectories(path) {
 
 function serveIndexWithRouting(file, req, res, next) {
   const routeTemplate = envRequire(file)
-
   const location = parsePath(req.url)
 
+  if (routeTemplate.handleRequest)
+    return routeTemplate.handleRequest(req, res, location).catch(error => {
+      reportServerError(error, req)
+      serveInternalServerError(error, req, res, next)
+    })
```

Comment:
With this version we expect `handleRequest` to be async. Is that ok? I think it is :-D

I was also thinking: do we want `location` to be the first argument so that it more closely mirrors the previous endpoints? The first step you most likely will do in `handleRequest` is check the location to see if you need to do anything special.